components/core/src/clp_s/metadata_uploader/TableMetadataManager.cpp

```
@@ -0,0 +1,101 @@
+#include "TableMetadataManager.hpp"
+
+#include <filesystem>
+
+namespace clp_s::metadata_uploader {
+TableMetadataManager::TableMetadataManager(
+        std::optional<clp::GlobalMetadataDBConfig> const& db_config
+) {
+    if (db_config.has_value()) {
+        m_table_metadata_db = std::make_unique<MySQLTableMetadataDB>(
+                db_config->get_metadata_db_host(),
+                db_config->get_metadata_db_port(),
+                db_config->get_metadata_db_username(),
+                db_config->get_metadata_db_password(),
+                db_config->get_metadata_db_name(),
+                db_config->get_metadata_table_prefix()
+        );
+        m_table_metadata_db->open();
+        m_output_type = OutputType::Database;
+    } else {
+        throw OperationFailed(ErrorCodeBadParam, __FILENAME__, __LINE__);
+    }
+}
+
+TableMetadataManager::~TableMetadataManager() {
+    if (m_output_type == OutputType::Database) {
+        m_table_metadata_db->close();
+    }
+}
+
+void TableMetadataManager::update_metadata(
+        std::string const& archive_dir,
+        std::string const& archive_id
+) {
+    m_table_metadata_db->init(archive_dir);
+
+    auto archive_path = std::filesystem::path(archive_dir) / archive_id;
+    std::error_code ec;
+    if (false == std::filesystem::exists(archive_path, ec) || ec) {
+        throw OperationFailed(ErrorCodeBadParam, __FILENAME__, __LINE__);
+    }
+
+    ArchiveReader archive_reader;
+    archive_reader.open(
+            clp_s::Path{.source = clp_s::InputSource::Filesystem, .path = archive_path.string()},
+            NetworkAuthOption{}
+    );
+
+    auto schema_tree = archive_reader.get_schema_tree();
+    auto field_pairs = traverse_schema_tree(schema_tree);
+    if (OutputType::Database == m_output_type) {
```

Do we want to throw on unknown OutputType?

---

The constructor should throw an exception